src/mono/wasm/wasm.proj

```diff
@@ -3,6 +3,8 @@
   <UsingTask TaskName="Microsoft.WebAssembly.Build.Tasks.RunWithEmSdkEnv" AssemblyFile="$(WasmAppBuilderTasksAssemblyPath)" />
 
   <PropertyGroup>
+    <EMSDK_PATH Condition="'$(EMSDK_PATH)' == ''">$(EMSDK)</EMSDK_PATH>
```

Comment:
What are the benefits at this stage? Are you using the wasmruntime packages produced from this branch? I would rather put my effort into making more things working

Reply:
> What are the benefits at this stage?

It's the same as with typescript code - less diff from upstream in upstream files. Each diff is a maintainace overhead.

Also, it has been our policy to not break "upstream build". There are two reasons for this:
a) Debugging broken build is much easier when you have a working baseline, by following the "what is it that we do different" lines of logic.
b) It protects against designs which would not work as an upstreamable change.

Reply:
I see. I don't think I can make it without at least adding some conditions in wasm.proj. Do you prefer to have such changes or should I just duplicate the wasm.proj?

Reply:
> don't think I can make it without at least adding some conditions in wasm.proj. Do you prefer to have such changes?

Yes; it looks like there shouldn't be that many.

Reply:
Would be enough to add conditions in wasm.proj based on `PrimaryRuntimeFlavor`?

Reply:
I don't think it should be `PrimaryRuntimeFlavor` (this really should be `Mono` for WASM, but isn't for historical reasons), but just `RuntimeFlavor` should/could work. The separate project is not strictly necessary, I was just thinking it could help to reduce the number of conditions required. If it's not necessary that's all the better.

Reply:
In https://github.com/dotnet/runtimelab/pull/2453/commits/993551519a976c59cec5d89d758527cc4ef5be40 I have tried to make the changes to wasm.proj smallest possible. I don't see how new project would minify them, but I'm open to suggestions